### PR TITLE
Migrate some modules in SimPPS to esConsumes

### PR DESCRIPTION
**Pull request by makortel**
#### PR description:

Part of #31061.

#### PR validation:

Code compiles.

### CODE REVIEW

**Comment by makortel** [comment]
@cmsbuild, please test

File: SimPPS/RPDigiProducer/plugins/RPDigiProducer.cc [hunk 1]
```diff
@@ -53,7 +52,7 @@ namespace CLHEP {
 class RPDigiProducer : public edm::EDProducer {
```
Comment (makortel) [inline SimPPS/RPDigiProducer/plugins/RPDigiProducer.cc:52]: This module needs to be migrated to some thread-efficient base class.

Is this module run in `runTheMatrix`? It is included in a list of modules being run that I extracted in last spring, but I don't see it being flagged as one preventing efficient use of concurrent lumis (that legacy EDModules imply) in https://github.com/cms-sw/cmssw/issues/25090.

**Comment by makortel** [comment]
And now I noticed the migration of `RPDigiProducer` is incomplete (the report in IBs missed this call). It uses `RPDetDigitizer` in
https://github.com/cms-sw/cmssw/blob/9b195809db69293bb06e902ba80214f52bd3a03b/SimPPS/RPDigiProducer/plugins/RPDigiProducer.cc#L174-L176
that uses `RPDisplacementGenerator` in
https://github.com/cms-sw/cmssw/blob/9b195809db69293bb06e902ba80214f52bd3a03b/SimPPS/RPDigiProducer/plugins/RPDetDigitizer.cc#L27
that gets data from `EventSetup`, .e.g
https://github.com/cms-sw/cmssw/blob/9b195809db69293bb06e902ba80214f52bd3a03b/SimPPS/RPDigiProducer/plugins/RPDisplacementGenerator.cc#L26-L29
)

This needs to be changed to something that allows declaring the consumed EventSetup products to be declared in the `RPDetDigitizer` constructor. It is far from clear to me what would be the best way to do that (e.g. is the map from `RPDetId` to `RPDetDigitizer` really necessary).

**Comment by fabferro** [comment]
@mundim : can you check? Help in the RP digitizers is needed too. Thanks!

**Comment by clemencia** [comment]
Hello, @mundim  asked me to take care of this so he can focus on the fullsim

**Comment by makortel** [comment]
@cms-sw/simulation-l2 Could you review and sign this PR even if it not complete for esConsumes in `RPDigiProducer`? The necessary work could come in later PR(s) (that I suppose would be easier to deal with given that @clemencia would work on that)

**Comment by civanch** [comment]
@makortel , @fabferro , @clemencia , I would propose that tomorrow morning I will sign this PR and make issue for PPS development. If you disagree, please, let me know.

**Comment by civanch** [comment]
+1

**Comment by perrotta** [comment]
+1

**Pull request merged**